### PR TITLE
feat: add monthly rotations UI

### PR DESCRIPTION
**Pull request by allending313**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Adds the monthly rotation features to the UI. Users can now select the monthly rotation option when creating or editing a rotation. Includes warning when handoff date is after the 28th as future handoff times may not be intuitive (example: jan 30th handoff date -> march 2nd, as February does not have a 30th date). This is done to keep the api rotation calculation and implementation clean.

**Which issue(s) this PR fixes:**
https://github.com/target/goalert/issues/3196

**Screenshots:**
Screenshot 2023-08-31 at 5 01 36 PM
Screenshot 2023-08-31 at 5 05 56 PM

**Describe any introduced user-facing changes:**
Additional rotation type option in create/edit rotation form.
Handoff summary is updated to reflect monthly rotation type. ("every month..." vs "every 2 months...")

### CODE REVIEW

**Review by Forfold — COMMENTED** [review]
Can we get the margin of the warning to match the errors?

Screenshot 2023-09-20 at 1 26 15 PM

Screenshot 2023-09-20 at 1 26 12 PM

File: web/src/app/rotations/RotationForm.tsx [hunk 1]
```diff
@@ -163,6 +173,12 @@ export default function RotationForm(props: RotationFormProps): JSX.Element {
               )
             }
           />
+          {handoffWarning && (
+            <Typography variant='body2' className={classes.handoffWarning}>
```
Comment (Forfold) [inline web/src/app/rotations/RotationForm.tsx:177]: you can apply styles directly now and I believe is the recommended way moving forward. it also comes with the theme object available as a callback function for every style:

```suggestion
            <Typography variant='body2' sx={{ color: (theme) => theme.palette.warning.main }}>
```

**Comment by allending313** [comment]
> Can we get the margin of the warning to match the errors?
> 
Does this work?

Screenshot 2023-09-21 at 11 49 12 AM

**Comment by Forfold** [comment]
yeah definitely!

**Review by Forfold — APPROVED** [review]

**Comment by mastercactapus** [comment]
Not sure if this is related to the monthly change or not, but seems odd to have negative values:
image

**Review by Forfold — APPROVED** [review]
LGTM!

**Review by Forfold — APPROVED** [review]

**Review by Forfold — DISMISSED** [review]

**Review by mastercactapus — DISMISSED** [review]

**Review by Forfold — APPROVED** [review]

**Pull request merged**